src/rust/engine/process_execution/src/remote.rs

```diff
@@ -372,8 +372,9 @@ impl super::CommandRunner for CommandRunner {
 
                             // take the grpc result and cancel the op if too much time has passed.
                             let elapsed = start_time.elapsed();
+                            let queue_buffer_time = std::time::Duration::from_secs(45);
```

Comment:
Maybe add a comment saying something along the lines of:
"We mostly trust the remote timeout that we set on the request; this timeout is only really here to make sure that if something goes really wrong and our connection hangs, or similar, we don't keep polling forever"